tests/build/build_templates.go

```diff
@@ -602,3 +615,41 @@ func getImageWithDigest(c *framework.ControllerHub, componentName, applicationNa
 	}
 	return fmt.Sprintf("%s@%s", url, digest), nil
 }
+
+func updateBuildPipelineSelector(kubeadminClient *framework.ControllerHub, testNamespace string) error {
+	//trueBool := true
+	customSourceBuildBundle := os.Getenv(constants.CUSTOM_SOURCE_BUILD_PIPELINE_BUNDLE_ENV)
+	Expect(customSourceBuildBundle).ShouldNot(BeEmpty())
+	if customSourceBuildBundle == "" {
+		return fmt.Errorf("source build bundle is empty")
+	}
+	bpsObj := &buildservice.BuildPipelineSelector{}
+	err := kubeadminClient.CommonController.KubeRest().Get(context.Background(), types.NamespacedName{Name: "build-pipeline-selector", Namespace: testNamespace}, bpsObj)
+	if err != nil {
+		return fmt.Errorf("error while getting build-pipeline-selector: %v", err)
+	}
+	found := false
+	for i := range bpsObj.Spec.Selectors {
+		selector := &bpsObj.Spec.Selectors[i]
+		if selector.Name == "Docker build" {
+			params := bpsObj.Spec.Selectors[i].PipelineRef.Params
+			for _, param := range params {
+				if param.Name == "bundle" {
+					found = true
+					break
+				}
+			}
+		}
+		if found {
+			selector.PipelineRef.Params[i].Value = *tektonpipeline.NewStructuredValues(customSourceBuildBundle)
```

Comment:
Is this correct? `i` is the index of `Selectors`, but here `i` is used to index a `Param`.

Reply:
Good catch. Fixed now.